CMakeLists.txt

```
@@ -20,59 +20,48 @@ set(CMAKE_CXX_EXTENSIONS OFF) #...without compiler extensions like gnu++11
 option(CMAKE_VERBOSE_MAKEFILE "Generate verbose Makefiles" OFF)
 include(GNUInstallDirs)
 
+# allow users to directly point to a kokkos install
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-find_package(Kokkos REQUIRED)
-
-option(Cabana_ENABLE_Serial "Build Cabana with Serial support" ON)
+set(Kokkos_INSTALL_PATH "" CACHE PATH "Kokkos installation location" )
+if(Kokkos_INSTALL_PATH)
+  find_package(Kokkos 3 PATHS ${Kokkos_INSTALL_PATH} NO_DEFAULT_PATH)
+endif()
+if(NOT Kokkos_FOUND)
+  find_package(Kokkos 3 REQUIRED)
+endif()
 
-option(Cabana_ENABLE_Pthread "Build Cabana with Pthread support" OFF)
-if( Cabana_ENABLE_Pthread )
-  find_package(Threads)
+# check user requirements
+option(Cabana_REQUIRE_SERIAL "Build Cabana with required Kokkos SERIAL support" OFF)
+if( Cabana_REQUIRE_SERIAL )
+  kokkos_check( DEVICES SERIAL )
 endif()
 
-option(Cabana_ENABLE_OpenMP "Build Cabana with OpenMP support" OFF)
-if( Cabana_ENABLE_OpenMP )
-  find_package(OpenMP)
+option(Cabana_REQUIRE_OPENMP "Build Cabana with required Kokkos OPENMP support" OFF)
+if( Cabana_REQUIRE_OPENMP )
+  kokkos_check( DEVICES OPENMP )
 endif()
 
-option(Cabana_ENABLE_Cuda "Build Cabana with Cuda support" OFF)
-if( Cabana_ENABLE_Cuda )
-  enable_language( CUDA )
+option(Cabana_REQUIRE_THREADS "Build Cabana with required Kokkos THREADS support" OFF)
+if( Cabana_REQUIRE_THREADS )
+  kokkos_check( DEVICES THREADS )
 endif()
 
-set(CABANA_SUPPORTED_DEVICES Serial Pthread OpenMP Cuda)
-set(ENABLE_DEVICES)
-foreach(_device ${CABANA_SUPPORTED_DEVICES})
-  if(Cabana_ENABLE_${_device})
-    string(TOUPPER ${_device} _uppercase_device)
-    if(NOT Kokkos_DEVICES MATCHES ".*${_uppercase_device}.*")
-      message(FATAL_ERROR "Kokkos built with devices '${Kokkos_DEVICES}', but Cabana with ${_device} requested.")
-    endif()
-    list(APPEND ENABLE_DEVICES "${_device}")
-  endif()
-endforeach()
-if(NOT ENABLE_DEVICES)
-  message(FATAL_ERROR "NO Cabana backend enabled")
+option(Cabana_REQUIRE_CUDA "Build Cabana with required Kokkos CUDA support" OFF)
+if( Cabana_REQUIRE_CUDA )
+  kokkos_check( DEVICES CUDA )
 endif()
-message(STATUS "Enable Devices: ${ENABLE_DEVICES}")
-
-# if we are using Cuda, also add a "CudaUVM" device to use that memory space
-# in the test suite. Also enable a Cabana UVM option so the tests get triggered.
-foreach(_device ${ENABLE_DEVICES})
-  if(_device STREQUAL Cuda)
-    list(APPEND CABANA_SUPPORTED_DEVICES CudaUVM)
-    set(Cabana_ENABLE_CudaUVM ON)
-  endif()
-endforeach()
 
 option(Cabana_ENABLE_TESTING "Build tests" OFF)
 if(Cabana_ENABLE_TESTING)
   enable_testing()
 endif()
 
-find_package(Doxygen)
-if(Doxygen_FOUND)
-  doxygen_add_docs(doxygen core/src)
+option(Cabana_ENABLE_DOXYGEN "Build doxygen" OFF)
```

```suggestion
option(Cabana_ENABLE_DOXYGEN "Build doxygen" ON)
```
Must default to `ON` if you want to prevent the previous behavior.  Also not advertised in the description.  This option feels unnecessary to me.

---

Another user request to reduce build times by making this optional - I can default this to `ON` though. Also documented in the PR description now.

---

This would only affect configuration and I doubt the user can tell the difference.  The documentation is only built if one run `make doxygen`, it is not part of the default build target.

---

Agreed.

---

OK - removing as well